spec/git_spec.rb

```diff
@@ -154,175 +154,202 @@
       remove_temp_repo(@local)
     end
-    
+
   end
-  
+
   context "cloning a bare repository" do
     before(:each) do
       remote = TEST_BARE_REPO_PATH
       @local  = get_new_temp_repo_path(true)
       @clone = RubyGit.clone(remote, @local, is_bare: true)
     end
-    
+
     it "is be bare" do
       expect(@clone).to be_bare
     end
-    
+
     after(:each) do
       remove_temp_repo(@local)
     end
   end
-  
+
   context "patching" do
     before(:each) do
       @temp_repo_path = create_temp_repo(TEST_REPO_PATH)
       @repo = Repo.new(@temp_repo_path)
     end
-    
+
     it "applies a patch from a String" do
       patch = fixture('postpatriarchialist.patch')
       result = @repo.git.apply_patch(patch)
       expect(result).to have(1).changed_file
       expect(result.first).to match /postpatriarchialist.txt/
     end
-    
+
     it "applies a patch from a file" do
       patch = File.join(File.dirname(__FILE__), 'fixtures', 'postpatriarchialist.patch')
       result = @repo.git.apply_file(patch)
       expect(result).to have(1).changed_file
       expect(result.first).to match /postpatriarchialist.txt/
     end
-    
+
     after(:each) do
       remove_temp_repo(@temp_repo_path)
     end
-    
+
   end
-  
-  describe "push and pull" do 
+
+  describe "push, pull and fetch" do
     before(:each) do
       @temp_repo_path = create_temp_repo(TEST_REPO_PATH)
       @remote = Repo.new(@temp_repo_path)
       @local = @remote.git.clone(@remote.path, get_new_temp_repo_path)
     end
-    
+
     context "pulling from another repository" do
       before(:each) do
         File.open(File.join(@remote.path, "materialist.txt"), 'a') {|file| file.write("\n Beautiful materialist.") }
         @remote.add("materialist.txt")
         @commit = @remote.commit("Making a change for pushing and pulling specs")
       end
-    
+
       it "pulls commits from a local clone" do
         expect(@local.commits).to have(8).commits
         @local.git.pull("origin", "master")
         expect(@local.commits).to have(9).commits
       end
-      
+
       it "pulls commits from a local clone with rebase" do
         expect(@local.commits).to have(8).commits
         @local.git.pull(nil, nil, rebase: true)
         expect(@local.commits).to have(9).commits
       end
-    
+
       it "pulls commits from a local clone with credentials" do
         expect(@local.commits).to have(8).commits
         @local.git.pull(nil, nil, username: 'rspec', password: 'Hahmeid7')
         expect(@local.commits).to have(9).commits
       end
-    
+
+    end
+
+    context "fetching from another repository" do
+      before(:each) do
+        @remote.create_branch('a_new_branch')
+        @remote.checkout('a_new_branch')
+        File.open(File.join(@remote.path, "materialist.txt"), 'a') {|file| file.write("\n Beautiful materialist.") }
+        @remote.add("materialist.txt")
+        @commit = @remote.commit("Making a change for fetching specs")
```

Comment:
Really, the commit itself is unnecessary, but it does make the ref for the two branches different. I'll remove the assigning of @commit.